### PR TITLE
Paginate and filter storage client deals list

### PR DESCRIPTION
**Pull request by aarshkshah1992**
For #4185 .

### TODO
- [ ] Merge https://github.com/filecoin-project/go-fil-markets/pull/512 and release
- [ ] Merge https://github.com/filecoin-project/lotus/pull/5853
- [ ] Update deps.

### CODE REVIEW

**Comment by aarshkshah1992** [comment]
@dirkmc @jennijuju 

**All possible options with which we can filter**

```
arshshah@Aarshs-MacBook-Pro lotus % lotus client list-deals --help                                                      
NAME:
   lotus client list-deals - List storage market deals

USAGE:
   also supports pagination, please see supported options for more details

CATEGORY:
   STORAGE

OPTIONS:
   --verbose, -v                 print verbose deal details (default: false)
   --color                       use color in display output (default: true)
   --watch                       watch deal updates in real-time, rather than a one time list (default: false)
   --hide-failed                 hide failed/failing deals (default: false)
   --creation-time-offset value  minimum creation time offset of the deals to include in the deal list page (default: no-op)
   --limit value                 number of deals to show in the deal list page (default: 50)
   --min-start-epoch value       minimum start epoch of the deals to include in the deal list page (default: no-op)
   --max-end-epoch value         maximum end epoch of the deals to include in the deal list page (default: no-op)
   --help, -h                    show help (default: false)

```

**All Deals**

```
aarshshah@Aarshs-MacBook-Pro lotus % lotus client list-deals -v

Created                    DealCid                                                      DealId  Provider  State                         On Chain?       Slashed?  PieceCID                                                          Size       Price                     Duration  TransferChannelID                                                                                                              TransferStatus  Verified  Message
2021-03-26T13:24:26+05:30  bafyreibcacybtma7rdfycjkognrl6wdty3lkmrdvys4jtoqbiwi2uziuaa  10      f01000    StorageDealAwaitingPreCommit  Y (epoch 1629)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.000402281072888688 FIL  1300994   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884001  Completed       false     
2021-03-26T13:26:21+05:30  bafyreic33ishws3swia2j4wwajlavv7ullpii237pqygtm4o4fwttsdayu  11      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.00040227210578268 FIL   1300965   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884002  Completed       false     
2021-03-26T13:26:26+05:30  bafyreib4yuxx6jmtjonlfdvpi3wk7vmvtvhi6d6teo2qhqscpkznt7gvbq  13      f01000    StorageDealActive    Y (epoch 1628)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.000402271796572128 FIL  1300964   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884004  Completed       false     
2021-03-26T13:26:31+05:30  bafyreibj47awsvzkblc4r6xln7elgh5r6gr4ftbvqo67i3sbjyfhk75mr4  14      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.000402271487361576 FIL  1300963   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884005  Completed       false     
2021-03-26T13:26:34+05:30  bafyreignj7x7smarnhd7ndbgrsi6xtwwnanxndhtww7fsc2tadnnrwnhru  12      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.000402271178151024 FIL  1300962   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884003  Completed       false 
```

**Just show the first two deals**

```
aarshshah@Aarshs-MacBook-Pro lotus % lotus client list-deals -v --limit=2

Created                    DealCid                                                      DealId  Provider  State                         On Chain?       Slashed?  PieceCID                                                          Size       Price                     Duration  TransferChannelID                                                                                                              TransferStatus  Verified  Message
2021-03-26T13:24:26+05:30  bafyreibcacybtma7rdfycjkognrl6wdty3lkmrdvys4jtoqbiwi2uziuaa  10      f01000    StorageDealAwaitingPreCommit  Y (epoch 1629)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.000402281072888688 FIL  1300994   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884001  Completed       false     
2021-03-26T13:26:21+05:30  bafyreic33ishws3swia2j4wwajlavv7ullpii237pqygtm4o4fwttsdayu  11      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         baga6ea4seaqkmjd7eh6ypntkvtnfp65fjgqhhta44vik7ths7hkrrp32scruujy  1.984 KiB  0.00040227210578268 FIL   1300965   12D3KooWHxkZnb2UHS23QAE6Hs2GptYQpziQXaHwpHjnZGHdXtkf-12D3KooWGBzFUDEw14UtJg1MeYrN5b8UNnyNBgqjumrB9p3wfqQ7-1616743172515884002  Completed       false     
```

**Show deals after a certain time offset**

```
aarshshah@Aarshs-MacBook-Pro lotus % lotus client list-deals --creation-time-offset="2021-03-26T13:26:21+05:30"
DealCid      DealId  Provider  State                         On Chain?       Slashed?  PieceCID     Size       Price                     Duration  Verified  
...wttsdayu  11      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         ...2scruujy  1.984 KiB  0.00040227210578268 FIL   1300965   false     
...znt7gvbq  13      f01000    StorageDealActive             Y (epoch 1628)  N         ...2scruujy  1.984 KiB  0.000402271796572128 FIL  1300964   false     
...fhk75mr4  14      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         ...2scruujy  1.984 KiB  0.000402271487361576 FIL  1300963   false     
...nnrwnhru  12      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         ...2scruujy  1.984 KiB  0.000402271178151024 FIL  1300962   false
```

**Show the first two deals after a certain time offset**

```
aarshshah@Aarshs-MacBook-Pro lotus % lotus client list-deals --creation-time-offset="2021-03-26T13:26:21+05:30" --limit=2
DealCid      DealId  Provider  State                         On Chain?       Slashed?  PieceCID     Size       Price                     Duration  Verified  
...wttsdayu  11      f01000    StorageDealAwaitingPreCommit  Y (epoch 1630)  N         ...2scruujy  1.984 KiB  0.00040227210578268 FIL   1300965   false     
...znt7gvbq  13      f01000    StorageDealActive             Y (epoch 1628)  N         ...2scruujy  1.984 KiB  0.000402271796572128 FIL  1300964   false   
```

File: cli/client.go [hunk 1]
```diff
@@ -1584,9 +1624,7 @@ func outputStorageDeals(ctx context.Context, out io.Writer, full lapi.FullNode,
 
 	var deals []deal
 	for _, localDeal := range localDeals {
-		if showFailed || localDeal.State != storagemarket.StorageDealError {
```
Comment (dirkmc) [inline cli/client.go:1587]: nit: let's remove the `showFailed` parameter from `outputStorageDeals()`

Comment (aarshkshah1992) [reply cli/client.go:1587]: This is done.

**Review by dirkmc — APPROVED** [review]
LGTM 🚀

**Comment by magik6k** [comment]
Built in markets are now deprecated in lotus, this feature won't land

**Pull request closed without merging**